utest/parsing/test_lexer.py

```diff
@@ -1816,5 +1816,149 @@ def _verify(self, data, expected, test=False):
         assert_tokens(data, expected, data_only=True)
 
 
+class TestContinue(unittest.TestCase):
+
+    def test_in_keyword(self):
+        data = '    CONTINUE'
+        expected = [(T.CONTINUE, 'CONTINUE', 3, 4),
+                    (T.EOS, '', 3, 12)]
+        self._verify(data, expected)
+
+    def test_in_test(self):
+        # This is not valid usage but that's not recognized during lexing.
+        data = '    CONTINUE'
+        expected = [(T.CONTINUE, 'CONTINUE', 3, 4),
+                    (T.EOS, '', 3, 12)]
+        self._verify(data, expected, test=True)
+
+    def test_in_if(self):
+        data = '''\
+    FOR    ${x}    IN    @{STUFF}
+        IF    True
+            CONTINUE
+        END
+    END
+'''
+        expected = [(T.FOR, 'FOR', 3, 4),
+                    (T.VARIABLE, '${x}', 3, 11),
+                    (T.FOR_SEPARATOR, 'IN', 3, 19),
+                    (T.ARGUMENT, '@{STUFF}', 3, 25),
+                    (T.EOS, '', 3, 33),
+                    (T.IF, 'IF', 4, 8),
+                    (T.ARGUMENT, 'True', 4, 14),
+                    (T.EOS, '', 4, 18),
+                    (T.CONTINUE, 'CONTINUE', 5, 12),
+                    (T.EOS, '', 5, 20),
+                    (T.END, 'END', 6, 8),
+                    (T.EOS, '', 6, 7)
+                    (T.END, 'END', 7, 4),
+                    (T.EOS, '', 6, 7)]
+        self._verify(data, expected)
+
+    def test_in_for(self):
+            data = '''\
+    FOR    ${x}    IN    @{STUFF}
+        CONTINUE
+    END
+'''
+            expected = [(T.FOR, 'FOR', 3, 4),
+                    (T.VARIABLE, '${x}', 3, 11),
+                    (T.FOR_SEPARATOR, 'IN', 3, 19),
+                    (T.ARGUMENT, '@{STUFF}', 3, 25),
+                    (T.EOS, '', 3, 33),
+                    (T.END, 'END', 7, 4),
+                    (T.EOS, '', 6, 7)]
+            self._verify(data, expected)
+
+    def _verify(self, data, expected, test=False):
+        if not test:
+            header = '*** Keywords ***'
+            header_type = T.KEYWORD_HEADER
+            name_type = T.KEYWORD_NAME
+        else:
+            header = '*** Test Cases ***'
+            header_type = T.TESTCASE_HEADER
+            name_type = T.TESTCASE_NAME
+        data = f'{header}\nName\n{data}'
+        expected = [(header_type, header, 1, 0),
+                    (T.EOS, '', 1, len(header)),
+                    (name_type, 'Name', 2, 0),
+                    (T.EOS, '', 2, 4)] + expected
+        assert_tokens(data, expected, data_only=True)
+
+
+class TestBreak(unittest.TestCase):
```

Comment:
Parsing tests for `CONTINUE` and `BREAK` are so similar that I think it would be best to have a single class for them. Each test in that class could then test same functionality with both of them.

Reply:
I also thought same thing but word length is different so I wrote it in separate class. Need your suggestion here.

Reply:
Do you mean that `BREAK` and `CONTINUE` have different length and thus offset of the following tokens are different? There's ought to be only an `EOS` on the same line and its offset ought to be pretty easy to set correctly based on the token type. I imagine something like this could work:
```python
class TestBreakAndContinue(unittest.TestCase):
    def test_in_keyword(self):
        for value in 'BREAK', 'CONTINUE':
            data = f'   {value}'
            expected = [(value, value, 3, 4),
                        (T.EOS, '', 3, 4 + len(value)]
            self._verify(data, expected)
```

This can potentially add some extra complexity so having separate test classes is fine as well. They can possibly be combined later if needed.